### PR TITLE
ROX-21610: reduce resource requests for local deploy

### PR DESCRIPTION
**Pull request by stehessel**
## Description

Some quality of life changes to make deploying ACS to dev clusters more convenient.

* Added a new flag `LOCAL_DEPLOYMENT` to hard code local deployment options. Set `LOCAL_DEPLOYMENT=true` for `deploy-local.sh` scripts. This way the "local" deploy settings, like reduced resource requests, can be used for test clusters on infra et al, too.
* Reduced resource requests even more for local deploys.
* Removed CPU limit for local deploys to prevent unnecessary throttling.

### CODE REVIEW

File: deploy/common/central-db-dev-patch.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            cpu: 100m
+            memory: 1Gi
+      containers:
+      - name: central-db
+        resources:
+          limits:
+            cpu: null
+            memory: 4Gi
+          requests:
+            cpu: 100m
```
Comment (dhaus67) [inline deploy/common/central-db-dev-patch.yaml:20]: This one might be a bit too low for the DB, let's bump it to `250m`?

File: deploy/common/central-dev-patch.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  template:
+    spec:
+      containers:
+      - name: central
+        resources:
+          limits:
+            cpu: null
+            memory: 4Gi
+          requests:
+            cpu: 100m
```
Comment (dhaus67) [inline deploy/common/central-dev-patch.yaml:11]: Same here, I feel like the reduce to 1/10th of the original resource requests might be too steep.

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]
Should the patch files be `-local-patch.yaml` instead of `-dev-patch.yaml`?

**Comment by stehessel** [comment]
> Should the patch files be `-local-patch.yaml` instead of `-dev-patch.yaml`?

I dislike the "local" wording because I want a similar setting for non-local but still dev infra clusters. But I guess it's too late for that...

**Review by gavin-stackrox — APPROVED** [review]

**Review by SimonBaeumer — APPROVED** [review]

**Pull request merged**